### DISCUSSION TITLE
i3wm weird behaviour on workspace change with unity

### DISCUSSION THREAD

**Discussion by tiozzomaksim** [i3/i3#5721] [Q&A]
Hello everyone, I'll start by saying that I'm new to i3wm, I've installed it recently.
I have this problem every time I switch between any workspace to unity workspace I have to hover some sections to make them appear (check the video below). I tried to disable my compositor, which would be picom, and tried to fullscreen the apps but nothing worked out. I even tried to look around on the internet and check the documentation but couldn't find anything.

https://github.com/i3/i3/assets/36891554/c9898a7b-5d32-4e07-ae20-0549a14907e6

Here is some info about my system in case is needed:

**OS:** Fedora release 38 (Thirty Eight) x86_64
**Kernel:** 6.5.6-200.fc38.x86_64
**WM:** i3 version 4.22
**GPU Driver:** NVIDIA 535.104.05
**Compositor:** picom vgit-751f3
**xrand:** 1.5.2

**Comment by ScallyGames** [comment]
Can confirm, I'm struggling from the same issue. 
If I'm remembering correctly this used to be better with an older version of Unity, I believe some 2021 or 2022 version. 
Not sure if the issue is on the side of i3, Unity or some interaction between the two.

PS: looks like it might be on the side of Unity.

Related links:
https://forum.unity.com/threads/editor-repaint-issue-when-using-i3-window-manager.657265/
https://github.com/tesq0/i3-unity-fix

Annoyingly the proposed fix by tesq0 does not work for me since I have the separate Unity panels as floating windows to be able to manage them through i3 (which is really nice apart from the repaint issue).
Are there any other ways to force a repaint / quick focus on all windows on workspace switch?
